src/NuGet.Core/NuGet.Packaging/PackageExtractor.cs

```diff
@@ -401,6 +401,7 @@ public static class PackageExtractor
                 var targetNuspec = versionFolderPathResolver.GetManifestFilePath(packageIdentity.Id, packageIdentity.Version);
                 var targetNupkg = versionFolderPathResolver.GetPackageFilePath(packageIdentity.Id, packageIdentity.Version);
                 var hashPath = versionFolderPathResolver.GetHashPath(packageIdentity.Id, packageIdentity.Version);
+                var nupkgMetadataFilePath = versionFolderPathResolver.GetNupkgMetadataPath(packageIdentity.Id, packageIdentity.Version);
```

Comment:
We're already making sure of that at https://github.com/NuGet/NuGet.Client/pull/2425/files#diff-e71565361fa493b38ca90727d1882fceR417 

It will come to this point when package was found at global or fallback folders and then also it make sure to check for `nupkg.metadata` file before proceeding with extraction. We can discuss offline if there are more confusions.

Reply:
I'm talking about a different scenario. 

I'm talking about the scenario where someone has put the nupkg,metadata file in the package itself. 
Never extract that similar to the _rels and [metadata] folders.

Reply:
discussed offline, will skip it similar to current hash file.